packages/renderer/src/lib/container/ContainerDetails.svelte

```diff
@@ -22,87 +23,86 @@ import ContainerDetailsTtyTerminal from './ContainerDetailsTtyTerminal.svelte';
 import { ContainerGroupInfoTypeUI, type ContainerInfoUI } from './ContainerInfoUI';
 import ContainerStatistics from './ContainerStatistics.svelte';
 
-export let containerID: string;
-
-let container: ContainerInfoUI;
+interface Props {
+  containerID: string;
+}
 
-let detailsPage: DetailsPage;
+let { containerID }: Props = $props();
 
-let displayTty = false;
-// update current route scheme
-let currentRouterPath: string;
+const containerUtils = new ContainerUtils();
 
-onMount(() => {
-  const containerUtils = new ContainerUtils();
+let detailsPage: DetailsPage | undefined = $state();
+let displayTty = $state(false);
 
-  router.subscribe(route => {
-    currentRouterPath = route.path;
-  });
-
-  // loading container info
-  return containersInfos.subscribe(containers => {
-    const matchingContainer = containers.find(c => c.Id === containerID);
-    if (matchingContainer) {
-      container = containerUtils.getContainerInfoUI(matchingContainer);
-      // look if tty is supported by this container
-      window
-        .getContainerInspect(container.engineId, container.id)
-        .then(inspect => {
-          displayTty = (inspect.Config.Tty ?? false) && (inspect.Config.OpenStdin ?? false);
-          // if we comes with a / redirect to /logs or to /tty if tty is supported
-          if (currentRouterPath.endsWith('/')) {
-            if (displayTty) {
-              router.goto(`${currentRouterPath}tty`);
-            } else {
-              router.goto(`${currentRouterPath}logs`);
-            }
+// update current route scheme
+let currentRouterPath: string = $derived($router.path);
+let containers: ContainerInfo[] = $derived($containersInfos);
+let container: ContainerInfoUI | undefined = $derived.by(() => {
+  const matchingContainer = containers.find(c => c.Id === containerID);
```

Comment:
suggestion: return containerInfoUI before doing the window.getContainerInspect, move the window.getContainerInspect in another $effect block that will be executed when containerInfoUI changes